core/dbt/config/runtime.py

```diff
@@ -171,3 +173,172 @@ def get_metadata(self) -> ManifestMetadata:
             project_id=self.hashed_name(),
             adapter_type=self.credentials.type
         )
+
+
+class PoisonedCredentials(Credentials):
```

Comment:
Can we `s/Poisoned/Unset` here? I'm also open to:
 - InvalidConfig
 - NotPresentConfig
 - MissingConfig

Reply:
Sure, I don't care at all what we call it